plugins/uefi-capsule/fu-plugin-uefi-capsule.c

```diff
@@ -500,6 +500,16 @@ fu_plugin_uefi_capsule_load_config(FuPlugin *plugin, FuDevice *device)
 		fu_device_add_private_flag(device, FU_UEFI_DEVICE_FLAG_FALLBACK_TO_REMOVABLE_PATH);
 }
 
+static FuVolume *
+fu_plugin_uefi_capsule_get_default_esp(FuPlugin *plugin, GError **error)
+{
+	g_autoptr(GPtrArray) esp_volumes = NULL;
+	esp_volumes = fu_context_get_esp_volumes(fu_plugin_get_context(plugin), error);
+	if (esp_volumes == NULL)
+		return NULL;
+	return g_object_ref(g_ptr_array_index(esp_volumes, 0));
```

Comment:
This is now a heuristic buried in the plugin source (if more than one ESP pick the first one).
Consequently I think it's a good idea to do something.
Ideas:
1) emit a message to the journal when more than one ESP was detected here.
2) An error saying we don't know "which" to pick, and don't use one.
3) If more than one, but one is ESP type and the other is BDP, pick the ESP type one.

Reply:
> If more than one, but one is ESP type and the other is BDP, pick the ESP type one.

That's the current ordering of the fu_context_get_esp_volumes() i.e. ESP first, BDP next. I think a g_warning() is appropriate here. Fixed.